src/mon/MgrMonitor.cc

```diff
@@ -70,77 +70,31 @@ static ostream& _prefix(std::ostream *_dout, Monitor &mon,
 
 // the system treats always_on_modules as if they provide built-in functionality
 // by ensuring that they are always enabled.
-const static std::map<uint32_t, std::set<std::string>> always_on_modules = {
-  {
-    CEPH_RELEASE_OCTOPUS, {
-      "crash",
-      "status",
-      "progress",
-      "balancer",
-      "devicehealth",
-      "orchestrator",
-      "rbd_support",
-      "volumes",
-      "pg_autoscaler",
-      "telemetry",
-    }
-  },
-  {
-    CEPH_RELEASE_PACIFIC, {
-      "crash",
-      "status",
-      "progress",
-      "balancer",
-      "devicehealth",
-      "orchestrator",
-      "rbd_support",
-      "volumes",
-      "pg_autoscaler",
-      "telemetry",
-    }
-  },
-  {
-    CEPH_RELEASE_QUINCY, {
-      "crash",
-      "status",
-      "progress",
-      "balancer",
-      "devicehealth",
-      "orchestrator",
-      "rbd_support",
-      "volumes",
-      "pg_autoscaler",
-      "telemetry",
-    }
-  },
-  {
-    CEPH_RELEASE_REEF, {
-      "crash",
-      "status",
-      "progress",
-      "balancer",
-      "devicehealth",
-      "orchestrator",
-      "rbd_support",
-      "volumes",
-      "pg_autoscaler",
-      "telemetry",
-    }
-  },
-  {
-    CEPH_RELEASE_SQUID, {
-      "crash",
-      "status",
-      "progress",
-      "balancer",
-      "devicehealth",
-      "orchestrator",
-      "rbd_support",
-      "volumes",
-      "pg_autoscaler",
-      "telemetry",
-    }
-  },
+static const std::map<uint32_t, std::set<std::string>>& always_on_modules() {
+  static const std::set<std::string> octopus_modules = {
+    "crash",
+    "status",
+    "progress",
+    "balancer",
+    "devicehealth",
+    "orchestrator",
+#ifdef WITH_RBD
+    "rbd_support",
+#endif
+#ifdef WITH_CEPHFS
+    "volumes",
+#endif
+    "pg_autoscaler",
+    "telemetry",
+  };
+  static const std::map<uint32_t, std::set<std::string>> always_on_modules_map = {
+    { CEPH_RELEASE_OCTOPUS, octopus_modules },
+    { CEPH_RELEASE_PACIFIC, octopus_modules },
+    { CEPH_RELEASE_QUINCY, octopus_modules },
+    { CEPH_RELEASE_REEF, octopus_modules },
+    { CEPH_RELEASE_SQUID, octopus_modules },
```

Comment:
ACK. I'm assuming before the changes there was no differences in module list between the versions.

Reply:
Correct. This way it's obvious and concise, previously it was a copy-paste of the same list.